server/routes/api/import_detection/detection.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions that manage all column detection."""
+
+import routes.api.import_detection.detect_date as date_detector
+from routes.api.import_detection.detection_types import Column, MappingVal, MappedThing, MappingType
+from typing import Dict, Sequence, Union
+
+import json
+
+
+def _detect_date(
+        col_order: Sequence[Column],
+        cols_sampled: Dict[int, Sequence[str]]) -> Union[MappingVal, None]:
+
+    date_headers = []
+    date_cols = []
+    for col in col_order:
+        col_header = col.header
+        col_ind = col.column_index
+        vals = cols_sampled[col_ind]
+
+        if date_detector.detect_column_header(col_header):
+            date_headers.append(col)
+        if date_detector.detect_column_with_dates(vals):
+            date_cols.append(col)
+
+    # If both date_headers and date_cols are non-empty,
+    # return the date_headers.
+    # If date_headers is empty but date_cols has more
+    # than one column, return any (e.g. the first one).
+    if date_headers:
+        return MappingVal(type=MappingType.COLUMN_HEADER.value,
+                          column=None,
+                          place_property=None,
+                          place_type=None,
```

Comment:
nit: is it worth defaulting None-able fields to None.

Reply:
Done!